csm/core/controllers/information.py

```diff
@@ -77,3 +77,115 @@ async def post(self):
             Log.error(f"Error in checking compatability: {e}")
             raise CsmInternalError(f"{e}")
         return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology")
+class TopologyView(CsmView):
+    """
+    Complete CORTX topology will be provided
+    GET: Complete CORTX topology
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation for complete cortx topology."""
+        Log.info("Fetching cortx topology.")
+        # Call Cortx Information Service
+        response = await self._service.get_topology()
```

Comment:
Any JSON related errors needs to be catched? Or you have taken care in other classes?

Reply:
Taken care by csmresponse

Reply:
There is no try/catch block in services or plugin too. If the code throws an error due to unknown error what will happen?

Reply:
Added try and catch block